Homework9/Hw9/Services/ExpressionParser/ExpressionParserService.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System.Linq.Expressions;
+using static Hw9.ErrorMessages.MathErrorMessager;
+
+namespace Hw9.Services.ExpressionParser
+{
+    public class ExpressionParserService : IExpressionParserService
+    {
+        public const string Plus = "+";
+        public const string Minus = "-";
+        public const string Multiply = "*";
+        public const string Divide = "/";
+        public const string OpenBracket = "(";
+        public const string CloseBracket = ")";
+        public const string UnaryMinus = "_";
+
+        public bool CheckExpressionString(string? expression, out string errorMessage)
+        {
+            if (expression == null)
+            {
+                errorMessage = EmptyString;
+                return false;
+            }
+
+            var openBracketsStack = new Stack<string>();
+
+            var symbols = expression.Split(' ');
```

Comment:
Например так (к комментарию интерфейса)
```suggestion
            var symbols = FormatIntoCorrectExpressionString(expression).Split(' ');
```